### ISSUE TITLE
[bilibili.com] only download partial of videos

### ISSUE THREAD

**Issue by jirat-sumitanun** [yt-dlp/yt-dlp#3659]
### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I'm running yt-dlp version **2022.04.08** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Description

`yt-dlp https://www.bilibili.com/video/BV1Xx411P7Ks`
I try to download this playlist https://www.bilibili.com/video/BV1Xx411P7Ks
 but it only download only partial of its.
full video is around 2hrs but it only download 6 minutes

### Verbose log

```shell
[debug] Override config: ['-vU', 'https://www.bilibili.com/video/BV1Xx411P7Ks']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, err utf-8, pref cp1252 
[debug] yt-dlp version 2022.04.08 [7884ade65] 
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Python version 3.9.7 (CPython 64bit) - Windows-10-10.0.19043-SP0 
[debug] Checking exe version: ffprobe -bsfs
[debug] Checking exe version: ffmpeg -bsfs 
[debug] exe versions: ffmpeg 4.4-full_build-www.gyan.dev (setts), ffprobe 4.4-full_build-www.gyan.dev, phantomjs 2.1.1 
[debug] Optional libraries: Cryptodome-3.14.1, brotli-1.0.9, certifi-2021.10.08, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {} 
Latest version: 2022.04.08, Current version: 2022.04.08 
yt-dlp is up to date (2022.04.08) 
[debug] [BiliBili] Extracting URL: https://www.bilibili.com/video/BV1Xx411P7Ks 
[BiliBili] 1Xx411P7Ks: Grabbing original ID via API 
[BiliBili] 1755915: Downloading webpage 
[BiliBili] 1755915: Extracting videos in anthology 
[BiliBili] Downloading anthology 1755915 - add --no-playlist to just download video 
[download] Downloading playlist: infinite stratos 2季 声优见面会影像 
[BiliBili] playlist infinite stratos 2季 声优见面会影像: Downloading 3 videos
[download] Downloading video 1 of 3
[debug] [BiliBili] Extracting URL: https://www.bilibili.com/video/BV1Xx411P7Ks?p=1
[BiliBili] 1Xx411P7Ks: Grabbing original ID via API
[BiliBili] 1755915: Downloading webpage 
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[BiliBili] 1755915: Extracting videos in anthology 
[BiliBili] 1755915: Downloading tags 
[debug] Default format spec: bestvideo*+bestaudio/best 
[info] 1755915_part1: Downloading 1 format(s): 17 
[debug] Invoking downloader on "https://upos-hz-mirrorakam.akamaized.net/upgcxcode/49/50/2685049/2685049-10-15.flv?e=ig8euxZM2rNcNbRVhwdVhwdlhWdVhwdVhoNvNC8BqJIzNbfqXBvEqxTEto8BTrNvN0GvT90W5JZMkX_YN0MvXg8gNEV4NC8xNEV4N03eN0B5tZlqNxTEto8BTrNvNeZVuJ10Kj_g2UB02J0mN0B5tZlqNCNEto8BTrNvNC7MTX502C8f2jmMQJ6mqF2fka1mqx6gqj0eN0B599M=&uipk=5&nbs=1&deadline=1651839914&gen=playurlv2&os=akam&oi=2869230290&trid=b56c416bb21547549f5b457a167be10fu&platform=pc&upsig=a60531fab3a5c1242b6cdbf0a8afa1fa&uparams=e,uipk,nbs,deadline,gen,os,oi,trid,platform&hdnts=exp=1651839914~hmac=2a6676b94d7f5bcca799ee20d07956d7141c4a89c308724723c547e0bf4f3467&mid=0&bvc=vod&nettype=0&orderid=0,1&agrr=0&bw=73685&logo=80000000"
[debug] File locking is not supported on this platform. Proceeding without locking 
[download] Destination: output\昼场 [1755915_part1].flv 
[download] 100% of 25.93MiB in 00:24
[download] Downloading video 2 of 3 
[debug] [BiliBili] Extracting URL: https://www.bilibili.com/video/BV1Xx411P7Ks?p=2 
[BiliBili] 1Xx411P7Ks: Grabbing original ID via API
[BiliBili] 1755915: Downloading webpage 
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[BiliBili] 1755915: Extracting videos in anthology 
[BiliBili] 1755915: Downloading tags 
[debug] Default format spec: bestvideo*+bestaudio/best 
[info] 1755915_part2: Downloading 1 format(s): 20 
[debug] Invoking downloader on "https://upos-hz-mirrorakam.akamaized.net/upgcxcode/50/50/2685050/2685050-8-15.flv?e=ig8euxZM2rNcNbRVhwdVhwdlhWdVhwdVhoNvNC8BqJIzNbfqXBvEqxTEto8BTrNvN0GvT90W5JZMkX_YN0MvXg8gNEV4NC8xNEV4N03eN0B5tZlqNxTEto8BTrNvNeZVuJ10Kj_g2UB02J0mN0B5tZlqNCNEto8BTrNvNC7MTX502C8f2jmMQJ6mqF2fka1mqx6gqj0eN0B599M=&uipk=5&nbs=1&deadline=1651839940&gen=playurlv2&os=akam&oi=2869230290&trid=40f452c29b2447c7b5a3696c5803e883u&platform=pc&upsig=71fd02104587424b561172c929770a10&uparams=e,uipk,nbs,deadline,gen,os,oi,trid,platform&hdnts=exp=1651839940~hmac=c196882853b15286e42848ed62c9e1357df333435c588d7c1166d01dfe1cf742&mid=0&bvc=vod&nettype=0&orderid=0,1&agrr=0&bw=74618&logo=80000000"
[download] Destination: output\夜场 [1755915_part2].flv 
[download] 100% of 25.62MiB in 00:23
[download] Downloading video 3 of 3 
[debug] [BiliBili] Extracting URL: https://www.bilibili.com/video/BV1Xx411P7Ks?p=3
[BiliBili] 1Xx411P7Ks: Grabbing original ID via API
[BiliBili] 1755915: Downloading webpage 
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[BiliBili] 1755915: Extracting videos in anthology 
[BiliBili] 1755915: Downloading tags 
[debug] Default format spec: bestvideo*+bestaudio/best 
[info] 1755915_part3: Downloading 1 format(s): 1+0 
[debug] Invoking downloader on "https://upos-hz-mirrorakam.akamaized.net/upgcxcode/51/50/2685051/2685051_da3-1-30016.m4s?e=ig8euxZM2rNcNbdlhoNvNC8BqJIzNbfqXBvEqxTEto8BTrNvN0GvT90W5JZMkX_YN0MvXg8gNEV4NC8xNEV4N03eN0B5tZlqNxTEto8BTrNvNeZVuJ10Kj_g2UB02J0mN0B5tZlqNCNEto8BTrNvNC7MTX502C8f2jmMQJ6mqF2fka1mqx6gqj0eN0B599M=&uipk=5&nbs=1&deadline=1651839964&gen=playurlv2&os=akam&oi=2869230290&trid=5c175f4303b443fd8dee6a7a72be3587u&platform=pc&upsig=ea7c3edc6a5cf85c061b35bf24c273d3&uparams=e,uipk,nbs,deadline,gen,os,oi,trid,platform&hdnts=exp=1651839964~hmac=a063b6952dc1dab8a762b76d388f576f53c58129bed8f6d7dc908d8cc9c74d53&mid=0&bvc=vod&nettype=0&orderid=0,1&agrr=0&bw=29485&logo=80000000"
[download] Destination: output\特典 [1755915_part3].f1.mp4 
[download] 100% of 22.33MiB in 00:18
[debug] Invoking downloader on "https://upos-hz-mirrorakam.akamaized.net/upgcxcode/51/50/2685051/2685051_da3-1-30216.m4s?e=ig8euxZM2rNcNbdlhoNvNC8BqJIzNbfqXBvEqxTEto8BTrNvN0GvT90W5JZMkX_YN0MvXg8gNEV4NC8xNEV4N03eN0B5tZlqNxTEto8BTrNvNeZVuJ10Kj_g2UB02J0mN0B5tZlqNCNEto8BTrNvNC7MTX502C8f2jmMQJ6mqF2fka1mqx6gqj0eN0B599M=&uipk=5&nbs=1&deadline=1651839964&gen=playurlv2&os=akam&oi=2869230290&trid=5c175f4303b443fd8dee6a7a72be3587u&platform=pc&upsig=f73becab62de776571f9aea5b826fdef&uparams=e,uipk,nbs,deadline,gen,os,oi,trid,platform&hdnts=exp=1651839964~hmac=38075a9956077a31a06811acb498e6bf5aa6bdb984e6083301f6468636bca34a&mid=0&bvc=vod&nettype=0&orderid=0,1&agrr=0&bw=8208&logo=80000000"
[download] Destination: output\特典 [1755915_part3].f0.m4a 
[download] 100% of 6.22MiB in 00:00
[Merger] Merging formats into "output\特典 [1755915_part3].mp4"
[debug] ffmpeg command line: ffmpeg -y -loglevel "repeat+info" -i "file:output\特典 [1755915_part3].f1.mp4" -i "file:output\特典 [1755915_part3].f0.m4a" -c copy -map "0:v:0" -map "1:a:0" -movflags "+faststart" "file:output\特典 [1755915_part3].temp.mp4"
Deleting original file output\特典 [1755915_part3].f0.m4a (pass -k to keep) 
Deleting original file output\特典 [1755915_part3].f1.mp4 (pass -k to keep) 
[download] Finished downloading playlist: infinite stratos 2季 声优见面会影像
```
```

**Comment by flashdagger** [comment]
The video is split in multiple parts. You need to download all formats and then concatenate them with ffmpeg.
`yt-dlp https://www.bilibili.com/video/BV1Xx411P7Ks  --no-playlist  -f all -o "%(title)s.%(format_id)02d.%(ext)s"`

**Comment by lockmatrix** [comment]
Hi everyone,
I create a temporary fix, https://github.com/lockmatrix/yt-dlp/commit/3a5833b078735b35b7bdd0bcf8eacc60b31d7f57
But I dont know how to download flv_fragments,
maybe someone can finish it .

It temporary download ios mp4 instead.

**Comment by lockmatrix** [comment]
PR to fix it: 
https://github.com/yt-dlp/yt-dlp/pull/4221

**Comment by pukkandan** [comment]
@lockmatrix Instead of commenting on each issue, just write `closes #3659` in your PR description. Github will pick it up and link all the issues with it

**Comment by lockmatrix** [comment]
> Instead of commenting on each issue, just write closes #3659 in your PR description. Github will pick it up and link all the issues with it

I got it

**Comment by grqz** [comment]
Closed by 1d6ab17d0752ee9cf19e3e63c7dec7b600d3f228
